Mathlib/Probability/Distributions/Gaussian.lean

```diff
@@ -192,6 +223,108 @@ lemma rnDeriv_gaussianReal (μ : ℝ) (v : ℝ≥0) :
+lemma gaussianReal_map_add_const (y : ℝ) :
+    (gaussianReal μ v).map (· + y) = gaussianReal (μ + y) v := by
+  by_cases hv : v = 0
+  · simp only [hv, ne_eq, not_true, gaussianReal_zero_var]
+    exact Measure.map_dirac (measurable_id'.add_const _) _
+  let e : ℝ ≃ᵐ ℝ :=
+  { toFun := fun x ↦ x - y
+    invFun := fun x ↦ x + y
+    left_inv := fun x ↦ by simp
+    right_inv := fun x ↦ by simp
+    measurable_toFun := measurable_id'.add_const _
+    measurable_invFun := measurable_id'.add_const _}
+  have he' : ∀ x, HasDerivAt e ((fun _ ↦ 1) x) x := fun _ ↦ (hasDerivAt_id _).sub_const y
+  suffices (gaussianReal μ v).map e.symm = gaussianReal (μ + y) v by exact this
+  ext s' hs'
+  rw [MeasurableEquiv.gaussianReal_map_symm_apply hv e he' hs']
+  simp only [abs_neg, abs_one, MeasurableEquiv.coe_mk, Equiv.coe_fn_mk, one_mul, ne_eq]
+  rw [gaussianReal_apply_eq_integral _ hv hs']
+  simp_rw [gaussianPdfReal_sub _ y]
+
+/-- The map of a Gaussian distribution by multiplication by a constant is a Gaussian. -/
+lemma gaussianReal_map_const_mul (c : ℝ) :
+    (gaussianReal μ v).map (c * ·) = gaussianReal (c * μ) (⟨c^2, sq_nonneg _⟩ * v) := by
+  by_cases hv : v = 0
+  · simp only [hv, mul_zero, ne_eq, not_true, gaussianReal_zero_var]
+    exact Measure.map_dirac (measurable_id'.const_mul c) μ
+  by_cases hc : c = 0
+  · simp only [hc, zero_mul, ne_eq, abs_zero, mul_eq_zero]
+    rw [Measure.map_const]
+    simp only [ne_eq, measure_univ, one_smul, mul_eq_zero]
+    convert (gaussianReal_zero_var 0).symm
+    simp only [ne_eq, zero_pow', mul_eq_zero, hv, or_false]
+    rfl
+  let e : ℝ ≃ᵐ ℝ :=
+  { toFun := fun x ↦ c⁻¹ * x
+    invFun := fun x ↦ c * x
+    left_inv := fun x ↦ by simp [hc]
+    right_inv := fun x ↦ by simp [hc]
+    measurable_toFun := measurable_id'.const_mul _
+    measurable_invFun := measurable_id'.const_mul _}
+  have he' : ∀ x, HasDerivAt e ((fun _ ↦ c⁻¹) x) x := by
+    suffices ∀ x, HasDerivAt (fun x => c⁻¹ * x) (c⁻¹ * 1) x by rwa [mul_one] at this
+    exact fun _ ↦ HasDerivAt.const_mul _ (hasDerivAt_id _)
+  suffices (gaussianReal μ v).map e.symm = gaussianReal (c * μ) (⟨c^2, sq_nonneg _⟩ * v) by
+    exact this
+  ext s' hs'
+  rw [MeasurableEquiv.gaussianReal_map_symm_apply hv e he' hs']
+  simp only [MeasurableEquiv.coe_mk, Equiv.coe_fn_mk, ne_eq, mul_eq_zero]
+  rw [gaussianReal_apply_eq_integral _ _ hs']
+  swap
+  · simp only [ne_eq, mul_eq_zero, hv, or_false]
+    rw [← NNReal.coe_eq]
+    simp [hc]
+  simp_rw [gaussianPdfReal_inv_mul hc]
+  congr with x
+  suffices |c⁻¹| * |c| = 1 by rw [← mul_assoc, this, one_mul]
+  rw [abs_inv, inv_mul_cancel]
+  rwa [ne_eq, abs_eq_zero]
+
+variable {Ω : Type} [MeasureSpace Ω]
+
+/-- If `X` is a real random variable with Gaussian law with mean `μ` and variance `v`, then `X + y`
+has Gaussian law with mean `μ + y` and variance `v`. -/
+lemma gaussianReal_add_const {X : Ω → ℝ} (hX : Measure.map X ℙ = gaussianReal μ v)
+    (hXm : Measurable X) (y : ℝ) :
```

Comment:
You shouldn't need `hXm` here: if the image of `P` is a gaussian (and in particular nonzero), this means that `X` is aemeasurable to start with, and that should be enough to go on. Same thing in the next lemma.